### PR TITLE
fix: UnaryFactor Jacobian

### PR DESCRIPTION
**Pull request by nmahabadi**
* Fix Unaryfactor jacobian in LocalizationExample.cpp.
* Update gtsam document.

### CODE REVIEW

**Comment by nmahabadi** [comment]
@dellaert @varunagrawal this is a new PR with the latest from "develop". Let me know if this seems fine.

File: examples/LocalizationExample.cpp [hunk 1]
```diff
@@ -84,15 +84,17 @@ class UnaryFactor: public NoiseModelFactor1<Pose2> {
...
-    if (H) (*H) = (Matrix(2, 3) << 1.0, 0.0, 0.0, 0.0, 1.0, 0.0).finished();
+  Vector evaluateError(const Pose2& q, boost::optional<Matrix&> H = boost::none) const override {
+    // The measurement function for a GPS-like measurement h(q) which predicts the measurement (m) is h(q) = q, q = [qx qy qtheta]
+    // The error is then simply calculated as E(q) = h(q) - m:
+    // error_x = q.x - mx
+    // error_y = q.y - my
+    // Node's orientation reflects in the Jacobian, in tangent space this is equal to the right-hand rule rotation matrix
+    // H =  [ cos(q.theta)  -sin(q.theta) 0 ]
+    //      [ sin(q.theta)   cos(q.theta) 0 ]
+    const double cosTheta = cos(q.theta());
```
Comment (dellaert) [inline examples/LocalizationExample.cpp:95]: same comments as above of course :-)

**Review by dellaert — APPROVED** [review]

File: examples/LocalizationExample.cpp (hunk 1 above)
Comment (nmahabadi) [reply examples/LocalizationExample.cpp:95]: Right! it is smart to use the available feature, I will update the PR soon.

**Comment by nmahabadi** [comment]
@dellaert I updated the code/doc as you suggested. 
(I am developing a version of the UnaryFactor that takes the measurement heading into account, once finished I will suggest to have it as a stand-alone factor in gtsam/nonlinear for example. If you agree by this proposal I can come back later by a new feature)

**Comment by dellaert** [comment]
Merging failed. Try merging in develop?

**Comment by nmahabadi** [comment]
@dellaert I merged the latest develop. The merge seems ok.

**Pull request merged**